### PR TITLE
W-14792039 PDK final review

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 1]
```diff
@@ -12,19 +12,66 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
 
 == Before You Begin
 
-* xref:policies-pdk-publish-policies.adoc[Upload your Custom Policy to Exchange]
+* xref:policies-pdk-compile-policies.adoc[Compile your custom policy].
+* If applying you policy to Flex Gateway running in Local Mode, xref:policies-pdk-publish-policies.adoc[upload your custom policy to exchange].
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc]: I think there's a typo here: you => your

Also, were you referring here to Connected Mode?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 2]
```diff
@@ -12,19 +12,66 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
...
+  policyRef:
+    kind: Extension
+    name: <your_custom_policy>-implementation
+  config:
+    <custom-cofiguration-parameters>: "your parameter"
+----
++
+* `metadata.name`: Contains the custom policy instance name.
+* `spec.targetRef.name`: Contains the name of the API instance on which to apply the custom policy. The API instance must already have been defined and applied.
+* `spec.policyRef.name`: Contains the custom policy implementation to apply.
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc]: Just a suggestion, to get the policy ref name here, we could also cite the helper command we're presenting here: 
https://dev-docs-internal.kqa.msap.io/beta-pdk-ga/pdk/latest/policies-pdk-integration-tests#configure-your-custom-policy

The `make show-policy-ref-name`

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 3]
```diff
@@ -12,19 +12,74 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
...
 When you apply a custom policy in Connected Mode, the configuration parameters and metadata defined in `gcl.yaml` appear in the API Manager UI.
 
 To apply a custom policy in Connected Mode, see xref:gateway::policies-included-apply.adoc[Apply a Policy in Connected Mode]. 
 
-== Local Mode
+=== Update Policies in Connected Mode
+
+To update a custom policy in API Manager, you can either:
+
+* Delete the policy, and then re-apply the latest version of the policy. This automatically updates the policies binary file.
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc]: I would put this option as the second. And I would suggest that it is only for DEV policies (the policy versions uploaded with `make publish`). For policies applied on production APIs it is not desirable to, even for a short period, leave the API unprotected.

Actually... if we don't want to over explain, I would remove this alternative and simply instruct with the second.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 4]
```diff
@@ -12,19 +12,74 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
...
+kind: PolicyBinding
+metadata:
+  name: <custom-policy-id>
+spec:
+  targetRef:
+    kind: ApiInstance
+    name: <your-api-instance>
+  policyRef:
+    kind: Extension
+    name: <your_custom_policy>-implementation
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc]: This name won't be like this. It will be more like `<your_custom_policy>-v1-0-impl`

File: pdk/1.0/modules/ROOT/pages/policies-pdk-compile-policies.adoc [hunk 5]
```diff
@@ -4,38 +4,33 @@ include::_attributes.adoc[]
...
 
-You can compile the custom policy project after you complete all the steps in xref:policies-pdk-create-project.adoc[].
-
-You must compile the custom policy before you can deploy it to a Flex Gateway.
-
-After you compile your project, test your policy with the following tutorials:
-
-* xref:policies-pdk-debug-local.adoc[]
-* xref:policies-pdk-apply-policies.adoc[]
+Compile your custom policy to create the policies binary target files by using the `make build` command. You must compile your custom policy after every time you edit the `src/lib.rs` file before you can deploy the policy to a Flex Gateway.
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-compile-policies.adoc]: instead of saying "after every time you edit the `src/lib.rs` file". I would say "after every time you change the policy source code". Because the user may add other source code files inside `src/`. Also we could say "after every time you change the content of the `src/` folder".

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-headers.adoc [hunk 6]
```diff
@@ -51,7 +51,7 @@ async fn response_filter(response_state: ResponseState, _config: &Config) {
 }
 ----
 
-Envoy handles the method, scheme, path, authority, and status codes as headers. Modify the headers by accessing the request's `:method`, `:scheme`, `:path`, `:authority`, and `:status`. PDK provides methods for access:
+Envoy handles the method, scheme, path, authority, and status codes as headers. PDK provides the following methods to access and modify the request's `:method`, `:scheme`, `:path`, `:authority`, and `:status` headers:
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-headers.adoc]: some of them are not actually headers, but pseudo headers defined by Envoy. Eg: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers.

We could call them Envoy psuedo-headers I guess.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-headers.adoc [hunk 7]
```diff
@@ -95,11 +102,7 @@ Because Envoy uses the same buffer to share data from the headers and the body,
 . Read the headers and save the necessary values in a variable.
 . Read the body.
 
-You can read the headers and then the body in both the response and request.
-
-You cannot modify headers after reading the body. Complete all header modification before reading the body.
-
-For example:
+You can read the headers and then the body in both the response and request. However, you cannot modify headers after reading the body. Complete all header modification before reading the body, for example:
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-headers.adoc:105]: I added the comment in the code, but we could also add a short sentence here saying. If the body is going to be modified, it will be required to remove the incoming request content-length header value, since it will be no longer valid.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 8]
```diff
@@ -12,19 +12,82 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
 
 == Before You Begin
 
-* xref:policies-pdk-publish-policies.adoc[Upload your Custom Policy to Exchange]
+* xref:policies-pdk-compile-policies.adoc[Compile your custom policy].
+* If applying your policy to Flex Gateway running in Connected Mode, xref:policies-pdk-publish-policies.adoc[upload your custom policy to exchange].
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc]:
```suggestion
* If applying your custom policy to Flex Gateway running in Connected Mode, xref:policies-pdk-publish-policies.adoc[upload your custom policy to Exchange].
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 9]
```diff
@@ -12,19 +12,82 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
...
 
-To apply a custom policy in Local Mode, use either xref:gateway::flex-local-secure-api-with-basic-auth-policy.adoc[] or xref:gateway::flex-local-secure-api-with-auto-policy.adoc[] as an example. Replace the example configuration with your policy's configuration parameters.
+To create the local declarative files:
+
+. In the `target/wasm32-wasi/release` PDK directory, copy both of the following files:
++
+* `<your_custom_policy>_definition.yaml`
+* `<your_custom_policy>_implementation.yaml`
++
+. Paste the two files in the `/etc/mulesoft/flex-gateway/conf.d/` Flex Gateway configuration directory.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc:53]: This refers to the configuration directory in the production environment, separate from development environment? Does this need to be clear?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 10]
```diff
@@ -12,19 +12,82 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
...
++
+. Create a policy binding configuration file with a `.yaml` file extension to bind the policy to an API instance:
++
+* Give the file a custom name.
+* Save the file in the Flex Gateway configuration directory `/etc/mulesoft/flex-gateway/conf.d/custom`. This directory can contain multiple configuration files.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc:58]: I'm fairly certain this refers to the config directory in production environment, but I'm not 100% sure. Does it need to be made explicit?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-inject-parameters.adoc [hunk 11]
```diff
@@ -36,7 +35,7 @@ async fn request_filter(state: RequestState) -> RequestData<String>;
 async fn response_filter(state: ResponseState, RequestData(path): RequestData<String>);
 ----
 
-Instead of a function definition of:
+Instead of a function definition of defining the lambda as in the following example:
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-inject-parameters.adoc]: Maybe I'd rephrase this to a more natural expression:

"Instead of using a lambda as parameter:"

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 12]
```diff
@@ -12,19 +12,82 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
...
+  config:
+    <custom-cofiguration-parameters>: "your parameter"
+----
++
+* `metadata.name`: The custom policy instance name.
+* `spec.targetRef.name`: The name of the API instance of which to apply the custom policy. The API instance must already have been defined and applied.
+* `spec.policyRef.name`: The custom policy implementation ID. To get the policy ID, run the `make show-policy-ref-name` command from the policy's root directory.
+* `spec.config`: Your policy configuration data.
++
+. Save the file. The gateway automatically refreshes the configuration.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc:83]: So the gateway instance doesn't need to be manually restarted? Does "automatically refreshes" mean the same thing as automatically restarts?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc [hunk 13]
```diff
@@ -12,19 +12,82 @@ To learn more about defining configuration parameters, see xref:policies-pdk-cre
...
+
+=== Update Policies in Local Mode
+
+To update a custom policy for Flex Gateway running in Local Mode:
+
+* If you didn't edit the configuration parameters, replace your old `<your_custom_policy>_implementation.yaml` in you Flex Gateway configuration directory with the implementation file from your `target/wasm32-wasi/release` PDK directory.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-apply-policies.adoc]:
```suggestion
* If you did not edit the configuration parameters, replace your old `<your_custom_policy>_implementation.yaml` in your Flex Gateway configuration directory with the implementation file from your `target/wasm32-wasi/release` PDK directory.
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc [hunk 14]
```diff
@@ -37,7 +37,7 @@ pub trait TokenProvision {
 }
 ----
 
-=== Bearer Provider
+=== Extract a Bearer Provider Token
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc]: I think this title should be "Extract a token using the Bearer Provider"

File: pdk/1.0/modules/ROOT/pages/policies-pdk-architecture.adoc [hunk 15]
```diff
@@ -4,15 +4,15 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images
 
-The Policy Development Kit (PDK) consists of the:
+Flex Gateway Policy Development Kit (PDK) consists of the:
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-architecture.adoc]:
```suggestion
Flex Gateway Policy Development Kit (PDK) consists of:
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc [hunk 16]
```diff
@@ -59,7 +59,7 @@ async fn filter(
 // [...]
 ----
 
-=== Custom Provider
+=== Extract a Custom Provider Token
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc]: And here it woul be "Extract a token using the Custom Provider"

File: pdk/1.0/modules/ROOT/pages/policies-pdk-architecture.adoc [hunk 17]
```diff
@@ -4,15 +4,15 @@ include::_attributes.adoc[]
...
 
 * <<anypoint-cli-pdk-plugin>>: Generates and distributes the policy to Exchange
 * <<policy-template>>: Scaffolds new custom policies
 * <<sdk-building-tools, SDK Building Tools>>: Simplifies the development process
 
 == Anypoint CLI PDK Plugin
 
-The Anypoint CLI PDK plugin creates the PDK project and uploads policy assets to Exchange. After creating the project, PDK provides a simplied set of commands in a `Makefile` to complete all following development steps.
+The Anypoint CLI PDK plugin creates the PDK project and uploads policy assets to Exchange. After creating the project, PDK provides a simplified set of commands in a `Makefile` to complete all following development steps.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-architecture.adoc]: Uploading to Exchange only applies to Connected Mode. Does this need to be clarified here?
```suggestion
The Anypoint CLI PDK plugin creates the PDK project and uploads policy assets to Exchange (for Connected Mode only). After creating the project, PDK provides a simplified set of commands in a `Makefile` to complete all following development steps.
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-architecture.adoc [hunk 18]
```diff
@@ -34,7 +34,7 @@ Flex Gateway is built on Envoy which requires developed policies to be compatibl
 
 PDK provides building tools that abstract the developer from the event-oriented coding approach by using reactor and executor patterns. PDK provides the developer a way to code each incoming request in a linear method that results in: 
 
-* Less errors from using methods in the incorrect context
+* Less compile and runtime errors from using methods in the incorrect context
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-architecture.adoc]:
```suggestion
* Fewer compile and runtime errors from using methods in the incorrect context
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc [hunk 19]
```diff
@@ -436,7 +432,7 @@ async fn filter(
 }
 ----
 
-== Custom Claim Validation
+== Validate Custom Claims
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-jwt.adoc:435]: This section should be removed. I think @NachoRaik removed it from the source engineering document: https://docs.google.com/document/d/1dXpEO_53HhXV1GGBqJU0veLksWBMAPzrabOKnezygug/edit#heading=h.vymfgou3mfr

File: pdk/1.0/modules/ROOT/pages/policies-pdk-compile-policies.adoc [hunk 20]
```diff
@@ -4,38 +4,33 @@ include::_attributes.adoc[]
...
 
-The `build` command preforms the following actions:
+The `make build` command preforms the following actions:
 
-. Runs the `build-asset-files` script
-. Runs the `build` command to compile the policy to a `WebAssembly` binary file
-. Generates additional configuration files that are not generated by the `build-asset-files` command and are required to execute the policy
+. Runs the `make build-asset-files` script
+. Runs the `make build` command to compile the policy to a `WebAssembly` binary file
+. Generates additional configuration files that are not generated by the `make build-asset-files` command and are required to execute the policy
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-compile-policies.adoc]:
```suggestion
. Generates additional configuration files that are not generated by the `make build-asset-files` script and are required to execute the policy
```
Consistency with the usage in the first listed action.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-caching.adoc [hunk 21]
```diff
@@ -1,4 +1,4 @@
-= Sharing Data Between Workers and Caching
+= Sharing Data Between Workers and Configuring Caching
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-caching.adoc:1]: Nice!

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features.adoc [hunk 22]
```diff
@@ -160,3 +144,22 @@ async fn configure(launcher: Launcher, Configuration(bytes): Configuration) -> R
     Ok(())
 }
 ----
+
+[[provided-rust-source-code]]
+== Provided Rust Source Code
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features.adoc]: I would rename this title, to also express that this functionality is exposed by our rust libraries, like:

"Discovering the PDK features with provided source code"

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-http-request.adoc [hunk 23]
```diff
@@ -52,7 +52,9 @@ async fn configure(launcher: Launcher, Configuration(bytes): Configuration) -> R
 }
 ----
 
-You must define your service as a parameter in you schema definition to make HTTP requests. The following code snippet provides an example schema definition for an external service:
+You must define your service as a parameter in you schema definition to make HTTP requests. To learn more about defining parameters, see xref:policies-pdk-create-schema-definition.adoc[]
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-http-request.adoc]:
```suggestion
You must define your service as a parameter in your schema definition to make HTTP requests. To learn more about defining parameters, see xref:policies-pdk-create-schema-definition.adoc[]
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 24]
```diff
@@ -21,7 +21,7 @@ Before you can create a custom policy project, ensure that you have installed al
 [[create-a-new-policy-project]]
 == Create a New Policy Project
 
-To create a custom policy project, run the PDK CLI `create` command, replacing `<my-custom-policy>` with the name of your new policy:
+To create a custom policy project, run the PDK CLI `make create` command, replacing `<my-custom-policy>` with the name of your new policy:
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc]: Uhm in this case its not referring to a makefile goal.

It is referencing the `anypoint-cli-v4 pdk policy-project create`. So, the "make" part at least, shouldn't be there.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 25]
```diff
@@ -30,13 +30,13 @@ anypoint-cli-v4 pdk policy-project create --name <my-custom-policy>
 
 NOTE: If you receive a prompt requesting your group ID, see xref:policies-pdk-troubleshooting.adoc#group-id-error[Troubleshooting Group ID].
 
-Running the `create` command creates a custom policy project with the specified name. The project appears in a new directory or a directory specified using the `--output-dir` flag. The following tutorials refer to this as your project's root directory.
+Running the `make create` command creates a custom policy project with the specified name. The project appears in a new directory or a directory specified using the `--output-dir` flag. The following tutorials refer to this as your project's root directory.
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc]: Same here

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc [hunk 26]
```diff
@@ -30,13 +30,13 @@ anypoint-cli-v4 pdk policy-project create --name <my-custom-policy>
...
 
 The `assetID` of the new policy is the policy's name in lower case with the spaces amended together with a `-`. For example, a policy with the name `My Header Injection Policy` has the `assetID` of `my-header-injection-policy`. 
 
 The `groupID` of the new policy is the ID of the organization configured in Anypoint CLI.
 
-The `create` command supports the following flags:
+The `make create` command supports the following flags:
```
Comment (nledesma) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc]: and here

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-inject-parameters.adoc (hunk 11 above)
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-inject-parameters.adoc]: I don't think the meaning of the sentence is clear. Is my guess below correct?
```suggestion
The following example demonstrates defining the lambda of a function definition:
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-network-proxy.adoc [hunk 27]
```diff
@@ -15,29 +15,29 @@ NOTE: Depending on your network proxy, you might have to define the domains the
 [[configure-a-network-proxy]]
 == Configure a Network Proxy
 
-Configure the environment variables by running following command after replacing the variables with your information:
+Configure your `HTTP_PROXY` or `HTTPS_PROXY` environment variables by running one of the following commands after replacing the variables with your information:
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-network-proxy.adoc]:
```suggestion
Configure your `HTTP_PROXY` or `HTTPS_PROXY` environment variables by running one of the following commands after replacing the variable values with your information:
```
The variable values are what should be replaced, not the variables themselves.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-configure-network-proxy.adoc [hunk 28]
```diff
@@ -15,29 +15,29 @@ NOTE: Depending on your network proxy, you might have to define the domains the
...
 
-* `HTTPS_PROXY`:
+* `HTTP_PROXY`:
 +
 [source,ssh]
 ----
 export HTTPS_PROXY=<proxy-method>://<proxy-server>:<proxy-port>
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-configure-network-proxy.adoc]: This export example says "HTTPS_PROXY", but it's under "HTTP_PROXY".

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc (hunk 25 above)
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-create-project.adoc]:
```suggestion
Running the `make create` command creates a custom policy project with the specified name. The project appears in a default directory or a directory specified using the `--output-dir` flag. The following tutorials refer to this as your project's root directory.
```
Aren't both of them new directories? I assume the first one refers to a default directory of some kind?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 29]
```diff
@@ -52,7 +52,7 @@ The supported labels are:
 
 * `title`: The policy name that appears in the UI.
 * `description`: A brief explanation of the policy's behavior.
-* `category`: The category the policy appears under. Supported values are: `security`, `compliance`, `transformation`, `quality of service`, and `troubleshooting`.
+* `category`: The category the of the policy. Suggested categories are `security`, `compliance`, `transformation`, `quality of service`, and `troubleshooting`.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc]:
```suggestion
* `category`: The category the of the policy. The possible categories are `security`, `compliance`, `transformation`, `quality of service`, and `troubleshooting`.
```
"Suggested" implies that there are other possible categories. Is this the intent? If so then please disregard.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 30]
```diff
@@ -126,7 +126,7 @@ If you do not set a value for a required property when applying a policy, the pr
 
 For non-required properties, the API Manager UI suggests the configured default value for properties. If you do not specify the value for non-required properties, the property is not configured.
 
-In the following configuration table, if the policy was applied in API Manager, `user2` would appear as a suggested value for `username` but does not affect if the value is configured:
+In the following configuration table, if the policy was applied in API Manager, `user2` would appear as a suggested value for `username` but is not configured if the value is left blank:
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc]:
```suggestion
In the following configuration table, if the policy was applied in API Manager, `user2` would appear as a suggested value for `username` but is not configured if the value is omitted:
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc [hunk 31]
```diff
@@ -347,21 +347,23 @@ For more information about DataWeave support for Flex Gateway, see xref:gateway:
 [[use-custom-parameters-in-rust-source-code]]
 == Use Custom Parameters in Rust Source Code
 
-After you modify the `gcl.yaml` file, run the `build-asset-files` command to propagate the changes to the `src/generated/config.rs` rust file and the JSON schema for the UI:
+After you modify the `gcl.yaml` file, run one of the following commands to propagate the changes to the `src/generated/config.rs` rust file and the JSON schema for the UI:
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc]: I'm not seeing the preceding text, so maybe it's explained. But I'm confused about "JSON schema for the UI". Which UI? It it referring to API Manager UI? Or VS Code UI? Or something else?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-debug-local.adoc [hunk 32]
```diff
@@ -1,14 +1,14 @@
...
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images
 
-To test the functionality of your policy, Policy Development Kit (PDK) provides a pre-configured API instance located in the `<root-directory>/playground/config` directory.
+To test the functionality of your policy, Flex Gateway Policy Development Kit (PDK) provides a pre-configured API instance located in the `<root-directory>/playground/config` directory.
 
-After you successfully xref:policies-pdk-compile-policies.adoc[compile your custom policy project] and add a <<register, Flex Gateway registration file>> to the configuration directory, you can use the `run` command to deploy the API instance and your policy to begin testing.
+The PDK Debugging Playground only supports Flex Gateways running in Local Mode in a Docker container. If you plan to deploy or policy to a Flex Gateway running in Connected Mode or to a different platform, you should still test your policy in PDK by using Flex Gateways running in Local Mode in a Docker container for initial testing.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-debug-local.adoc]:
```suggestion
The PDK Debugging Playground only supports Flex Gateway running in Local Mode in a Docker container. If you plan to deploy a policy to Flex Gateway running in Connected Mode or to a different platform, you should still first test your policy in PDK by using Flex Gateway running in Local Mode in a Docker container.
```
I'm not sure what "or to a different platform" is referring to. You first mention "Connected Mode" - are you referring to its alternative: Local Mode? Or is this referring to something else? Production environment perhaps?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-debug-local.adoc [hunk 33]
```diff
@@ -1,14 +1,14 @@
...
 
-PDK only supports Flex Gateways running in a Docker container.
+IMPORTANT: You can run Flex Gateway in Local Mode on Docker in Windows only to develop and test your policies. However, Flex Gateway does not support Windows in production environments.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-debug-local.adoc:11]: Awesome. Thanks.

File: pdk/1.0/modules/ROOT/pages/policies-pdk-integration-tests.adoc [hunk 34]
```diff
@@ -6,14 +6,19 @@ endif::[]
 
 Flex Gateway Policy Development Kit (PDK) provides an automated testing framework to detect integration regressions in your custom policy.
 
+The integration testing framework provided by PDK only supports Flex Gateways running in Local Mode in a Docker container. If you plan to deploy or policy to a Flex Gateway running in Connected Mode or to a different platform, you should still test your policy in PDK by using Flex Gateways running in Local Mode in a Docker container for initial testing.
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-integration-tests.adoc]:
```suggestion
The integration testing framework provided by PDK only supports Flex Gateway running in Local Mode in a Docker container. If you plan to deploy or policy to Flex Gateway running in Connected Mode or to a different platform, you should still first test your policy in PDK by using Flex Gateway running in Local Mode in a Docker container.
```
Same question about "or to a different platform". Is this referring to a production environment?

File: pdk/1.0/modules/ROOT/pages/policies-pdk-prerequisites.adoc [hunk 35]
```diff
@@ -1,16 +1,16 @@
-= Reviewing Flex Gateway Policy Development Kit Prerequisites
+= PDK Prerequisites
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-prerequisites.adoc]:
```suggestion
= Reviewing PDK Prerequisites
```
This title should include a gerund so that it matches all the other titles.

File: pdk/1.0/modules/ROOT/nav.adoc [hunk 36]
```diff
@@ -1,7 +1,7 @@
-* xref:policies-pdk-overview.adoc[Policy Development Kit]
+* xref:policies-pdk-overview.adoc[PDK Overview]
 * xref:policies-pdk-release-notes.adoc[Release Notes]
 * xref:policies-pdk-architecture.adoc[Architecture Overview]
-* xref:policies-pdk-prerequisites.adoc[Reviewing Prerequisites]
+* xref:policies-pdk-prerequisites.adoc[]
 * xref:policies-pdk-configure-network-proxy.adoc[]
 * xref:policies-pdk-develop-custom-policies.adoc[Develop Custom Policies]
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/nav.adoc]:
```suggestion
* xref:policies-pdk-develop-custom-policies.adoc[]
```

File: pdk/1.0/modules/ROOT/nav.adoc [hunk 37]
```diff
@@ -1,7 +1,7 @@
...
 ** xref:policies-pdk-create-project.adoc[Create a New Project]
```
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/nav.adoc]:
```suggestion
** xref:policies-pdk-create-project.adoc[]
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc (hunk 31 above)
Comment (glenn-rodgers-sf) [reply pdk/1.0/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc]: API Manager UI added

File: pdk/1.0/modules/ROOT/pages/policies-pdk-integration-tests.adoc (hunk 34 above)
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-integration-tests.adoc]:
```suggestion
The integration testing framework provided by PDK only supports Flex Gateway running in Local Mode in a Docker container. If you plan to deploy or policy to Flex Gateway running in Connected Mode or to a different platform, you should still first test your policy in PDK by using Flex Gateway running in Local Mode in a Docker container.
```

File: pdk/1.0/modules/ROOT/pages/policies-pdk-debug-local.adoc (hunk 32 above)
Comment (chris-campbell9) [inline pdk/1.0/modules/ROOT/pages/policies-pdk-debug-local.adoc]:
```suggestion
The PDK Debugging Playground only supports Flex Gateway running in Local Mode in a Docker container. If you plan to deploy or policy to Flex Gateway running in Connected Mode or to a different platform, you should still first test your policy in PDK by using Flex Gateway running in Local Mode in a Docker container.
```

**Review by chris-campbell9 — APPROVED** [review]
